sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -4492,6 +4492,35 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
     CommentOnTable(createUnresolvedTable(ctx.multipartIdentifier, "COMMENT ON TABLE"), comment)
   }
 
+  /**
+   * Create an index, returning a [[CreateIndex]] logical plan.
+   * For example:
+   * {{{
+   * CREATE [index_type] INDEX [index_name] ON [TABLE] table_name (column_index_property_list)
```

Comment:
```suggestion
   * CREATE [index_type] INDEX index_name ON [TABLE] table_name (column_index_property_list)
```

index name is not optional